src/main/resources/application.conf

```diff
@@ -298,14 +298,20 @@ mantis {
     # When we receive a block that we don't have a parent for, it means we found a fork. To resolve it, we need to query the
     # same peer for previous blocks, until we find one that we have a parent for. This setting determines the batch size
     # of such query (number of blocks returned in a single query).
-    block-resolving-depth = 20
+    block-resolving-per-request = 20
 
     # size of the list that keeps track of peers that are failing to provide us with mpt node
     # we switch them to download only blockchain elements
     fastsync-block-chain-only-peers-pool = 100
 
     # time between 2 consecutive requests to peer when doing fast sync, this is to prevent flagging us as spammer
     fastsync-throttle = 2 seconds
+
+    # limit of length of branch in chain that will be resolved while doing regular sync
+    # if branch is longer resolution will be aborted because of risk of out of memory error
+    # too long branch may also indicate malicious peer, 24 because confirmation depth in ethereum is ~12 blocks
+    # so we allow branches of 2x confirmation depth
+    branch-max-depth-resolving = 24
```

Comment:
about `>=` I think we will not be making 1 redundant request, because we start resolving with 1 block header in the queue
we start resolving branches when new block header that we get do not have a parent in our chain